src/ngraph/runtime/reference/clamp.hpp

```diff
@@ -0,0 +1,52 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            void clamp(const T* arg, T* out, double min, double max, size_t count)
+            {
+                T min_t = static_cast<T>(min);
+                T max_t = static_cast<T>(max);
```

Comment:
Here I am casting min/max (type `double`) to the argument type T.  Information can be lost here.  For example, if min or max are negative and T is unsigned.  Or, if T is less than the size of min/max mantissa e.g. an 8 or 16 bit integer.  I can instead cast the argument to a `double` for comparison, but this doesn't solve the problem as I will still need to case min/max to type T in lines 38 and 42 below.

Reply:
Right -- you need to floor/ceiling for the int types. There's a template way to do that .